src/helper/horizon-rpc.ts

```diff
@@ -0,0 +1,195 @@
+import BigNumber from "bignumber.js";
+import { AssetType, Horizon, Server, ServerApi } from "stellar-sdk";
+
+export const BASE_RESERVE = 0.5;
+export const BASE_RESERVE_MIN_COUNT = 2;
+
+export interface Issuer {
+  key: string;
+  name?: string;
+  url?: string;
+  hostName?: string;
+}
+
+export interface NativeToken {
+  type: AssetType;
+  code: string;
+}
+
+export interface AssetToken {
+  type: AssetType;
+  code: string;
+  issuer: Issuer;
+  anchorAsset?: string;
+  numAccounts?: BigNumber;
+  amount?: BigNumber;
+  bidCount?: BigNumber;
+  askCount?: BigNumber;
+  spread?: BigNumber;
+}
+
+export type Token = NativeToken | AssetToken;
+
+export interface Balance {
+  token: Token;
+
+  // for non-native tokens, this should be total - sellingLiabilities
+  // for native, it should also subtract the minimumBalance
+  available: BigNumber;
+  total: BigNumber;
+  buyingLiabilities: BigNumber;
+  sellingLiabilities: BigNumber;
+}
+
+export interface AssetBalance extends Balance {
+  token: AssetToken;
+  sponsor?: string;
+}
+
+export interface NativeBalance extends Balance {
+  token: NativeToken;
+  minimumBalance: BigNumber;
+}
+
+export interface BalanceMap {
+  [key: string]: AssetBalance | NativeBalance;
+  native: NativeBalance;
+}
+
+export function getBalanceIdentifier(balance: Horizon.BalanceLine): string {
+  if ("asset_issuer" in balance && !balance.asset_issuer) {
+    return "native";
+  }
+  switch (balance.asset_type) {
+    case "credit_alphanum4":
+    case "credit_alphanum12":
+      return `${balance.asset_code}:${balance.asset_issuer}`;
+
+    case "liquidity_pool_shares":
+      return `${balance.liquidity_pool_id}:lp`;
+
+    default:
+      return "native";
+  }
+}
+
+export const makeDisplayableBalances = (
```

Comment:
This all comes from the wallet-sdk but I had a helluva time trying to build it for this project(esmodules), since this lib is deprecated I think it would be easiest to wait until the new TS wallet-sdk is ready and pull that in to replace these helpers eventually.

Reply:
Actually, it occurs to me, for a lot of the Horizon stuff account stuff - you could use `@typescript/wallet-sdk`. 

You could use `getInfo` here for example to get the ServerApi.AccountRecord: https://github.com/stellar/typescript-wallet-sdk/blob/b852646a6981e68f4ffc7a2b4cbb2b148315b1d0/src/walletSdk/Horizon/AccountService.ts#L49

Sorry I'm just thinking of this now! We should probably create a ticket for the backlog as a future improvement. We don't have an equivalent of `makeDisplayableBalances` so it's probably not the highest priority yet

Reply:
ah gotcha yeah let me see what the cross over is